docs/source/llms/dspy/notebooks/dspy_quickstart.ipynb

```diff
@@ -0,0 +1,927 @@
+    "\n",
+    "### This Demo\n",
+    "Below we create a simple program that demonstrates the power of DSPy. We will build a text classifier leveraging OpenAI. By the end of this tutorial, we will...\n",
+    "\n",
+    "1. Define a [dspy.Signature](https://dspy-docs.vercel.app/docs/building-blocks/signatures) and [dspy.Module](https://dspy-docs.vercel.app/docs/building-blocks/modules) to perform text classification.\n",
+    "2. Leverage [dspy.teleprompt.BootstrapFewShotWithRandomSearch](https://dspy-docs.vercel.app/api/optimizers/BootstrapFewShotWithRandomSearch) to compile our module so it's better at classifying our text.\n",
+    "3. Log the compiled model with MLflow.\n",
+    "4. Load the compiled model with MLflow and perform inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "58918566-6e93-4a2e-9a34-b0f56378885a",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "zsh:1: 0.2.5 not found\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "%pip install -U openai dspy>=2.5.1 mlflow>=2.17.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "7bd9e460-4c9c-4508-8801-6f29fcf2c8d2",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "# 0 - Setup\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "c27ec364-1195-447d-a9d5-f38defa6652e",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## 0.1 - Set Up LLM\n",
+    "\n",
+    "After installing the relevant dependencies, let's leverage a Databricks foundational model serving endpoint as our LLM of choice. Here, will leverage the open source Meta Llama 3.1 70B LLM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "3232fb03-f4be-490f-9179-0f2b71129196",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import dspy\n",
+    "\n",
+    "# Define your model. We will use OpenAI for simplicity\n",
```

Comment:
? Are we using OpenAI or Llama? 
Either way is fine; let's just be consistent in the text.